packages/app-bridge/src/types/Emitter.ts

```diff
@@ -13,12 +13,13 @@ import type { DocumentCategory } from './DocumentCategory';
 import type { BrandportalLink } from './BrandportalLink';
 import type { PrivacySettings } from './PrivacySettings';
 import type { Template } from './Template';
+import type { ThemeTemplate } from './ThemeTemplate';
 
 export type EmitterAction = 'add' | 'update' | 'delete';
 
 export type EmitterEvents = {
     'AppBridge:PageTemplateSettingsUpdated': { pageTemplateSettings: Record<string, unknown> };
-    'AppBridge:ThemeSettingsUpdated': { themeSettings: Record<string, unknown> };
+    'AppBridge:ThemeSettingsUpdated': { themeSettings: Record<ThemeTemplate, Record<string, unknown>> };
```

Comment:
Please explain the changes, `theme settings !== page template settings`

```
                 theme
                   |
         |----------------------|
theme settings    page template settings
                                 |
           |---------------------|------------------|
document page         cover page            library page
template settings     template settings     template settings
```

Reply:
`theme settings` are now per template, we have:
```
                                               theme
                                                    |
         |--------------------------------------------------|
theme settings                                            page template settings
        |                                                                                     |
 |-------------------|-----------|                       |---------------------|------------------|
documentPage     Cover        Library          document page         cover page            library page
                                                                     template settings     template settings     template settings
```
ThemeSettings are the settings applied to new items or when the user resets its value for each template. We'll have `ThemeSettings.<template>` for each template.

But still the user can still overwrite the specific document page, cover or library page settings with template settings. 

Dis I answer your question?

Reply:
The requirements have changed and we need to be able to define theme settings per template.

![image](https://github.com/Frontify/brand-sdk/assets/2899490/633be8ac-64c2-4b02-981e-bb70d80c00f1)

Reply:
@mike85, with defining "theme settings per template" do that mean you can have one template defining the settings for multiple themes? or is it one template defined settings for one theme?

Reply:
it seems something is off to me with the diagram representation: why are `theme settings` and `page template settings` siblings? Shouldn't `theme settings` be a sibling of `template settings`? 🤔 Essentially the settings of a theme and a template should define the same properties

```
                              theme
                                |
                             settings     <- template settings plugs in here
                                |           with same underlaying child logic
                                |
                             document
                                |
           |--------------------|------------------|
        cover                  page            library 
```

Also conceptually it seems we have a domain entity naming problem: document is broader than page. A document can have multiple pages, not the other way around.

Reply:
The flow of settings is as follow. We have sets of settings per template and then settings for individual items.
These settings are overwritten as the flow goes down, taking preference the latest.

```
  |                   Default Theme Settings                   <--- hardcoded in the theme
  |         (apply to all Cover, DocumentPage, Library)
 \ /                            |
  |                             |
  |                 Template Theme Settings                    <--- Editable by the Editor
  |         (apply to all Cover, DocumentPage, Library)
 \ /                            |
  |                             |
  |                     Template Settings                      <--- Editable by the Editor
  |         (individual Cover, DocumentPage or Library)
```

Reply:
Honestly I don't see the purpose of the "Template Theme Settings", but I may be missing something. There's nothing that `Template Settings` wouldn't answer that a `Template Theme Settings` does. To me it seems that `Template Theme Settings` shouldn't be a feature itself but rather a data migration applied on `Template Settings`.

Reply:
`Template Theme Settings` makes it easier to create/reset pages/libraries with the appropriated settings with no further actions. Once these `Template Theme Settings` are set, any new page inherits them. Otherwise the editor either has to set them for every individual page/library or has to use `apply-to` every time something changes. 

`Template Settings` only applies to individual pages/libraries not to all of their kind, `Template Theme Settings` does that.